File: ruby/reverse_words_in_a_string.rb
# @param {String} s
# # @return {String}
def reverse_words(s)
  arr = s.split
  (arr.size >> 1).times do |i|
    arr[i], arr[-(i+1)] = arr[-(i+1)], arr[i]
  end
  return arr.join(" ")
end

File: python/reverseWordsInAString.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        arr = s.strip().split()
        n = len(arr)
        for i in range(n >> 1):
            arr[i], arr[n - i - 1] = arr[n - i - 1], arr[i]
        return " ".join(arr)

if __name__ == "__main__":
    s = Solution()
    print s.reverseWords("the sky is blue")
